source/com.microsoft.tfs.client.common/src/com/microsoft/tfs/client/common/credentials/CredentialsHelper.java

```diff
@@ -161,4 +183,103 @@ public static TFSConnection getVstsRootConnection(final TFSConnection connection
 
         return vstsConnection;
     }
+
+    public static PatCredentials getOAuthCredentials(final URI serverURI) {
+        final Authenticator authenticator;
+        final Options options = Options.getDefaultOptions();
+        options.patGenerationOptions.displayName = PatCredentials.USERNAME_FOR_CODE_ACCESS_PAT;
+        options.patGenerationOptions.tokenScope = VsoTokenScope.AllScopes;
+
+        if (serverURI != null) {
+            log.debug("Interactively retrieving credential based on oauth2 flow for " + serverURI.toString()); //$NON-NLS-1$
```

Comment:
Those logging and comments were based on "persisting" flag in the tf clc flow.  Maybe we should update it?  I don't think the decision is based on persistence anymore.